loader/directory_refresher.go

```diff
@@ -3,17 +3,40 @@ package loader
 import "path/filepath"
 
 type DirectoryRefresher struct {
-	currDir string
+	currDir  string
+	watchOps map[FileSystemOp]struct{}
```

Comment:
nit: while it takes up slightly more memory, making this `map[FileSystemOp]bool` would let you look things up directly and leverage the default value, letting you drop the two-valued map lookup code.
```go
func (d *DirectoryRefresher) ShouldRefresh(path string, op FileSystemOp) bool {
  watchOps := d.watchOps
  if watchOps == nil {
        watchOps = defaultFileSystemOps
  }
  return filepath.Dir(path) && watchOps[op]
}
```

Reply:
Sounds good to me